pygbm/grower.py

```diff
@@ -70,8 +71,13 @@ def __init__(self, features_data, all_gradients, all_hessians,
         if not features_data.flags.f_contiguous:
             warnings.warn("Binned data should be passed as Fortran contiguous"
                           "array for maximum efficiency.")
+        if isinstance(n_bins_per_feature, int):
+            n_bins_per_feature = np.array(
+                [n_bins_per_feature] * features_data.shape[1],
+                dtype=np.uint32)
```

Comment:
I would pass `n_bins_per_features=None` and `max_bins=256` to the constructor and then:

```python
if n_bins_per_features is None:
    n_bins_per_features = max_bins

if isinstance(n_bins_per_feature, int):
    n_bins_per_feature = np.array(
        [n_bins_per_feature] * features_data.shape[1],
        dtype=np.uint32)
```